backend/src/errorHandler.ts

```diff
@@ -0,0 +1,28 @@
+import { Logger } from "pino";
+
+import logger from "./logger";
+
+class ErrorHandler {
+    private readonly logger: Logger;
+
+    constructor({ logger }: { logger: Logger }) {
+        this.logger = logger;
+    }
+
+    public async handleExceptionError(error: Error) {
+        this.logger.error(error, 'Uncaught exception: ');
+        // TODO: add monitoring
+        process.exit(1);
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    public async handleRejectionError(reason: any) {
+        this.logger.error(reason, 'Unhandled rejection: ');
+        // TODO: add monitoring
+        process.exit(1);
+    }
+}
+
+const errorHandler = new ErrorHandler({ logger });
```

Comment:
Honestly I extremely dislike such global instances. Same about logger BTW.

Reply:
just this then?
```
process.on('uncaughtException', (error: Error) => {
  logger.error(error, 'Uncaught exception: ');
  // TODO: add monitoring
  process.exit(1);
});

process.on('unhandledRejection', (reason) => {
  logger.error((reason as any), 'Unhandled rejection: ');
  // TODO: add monitoring
  process.exit(1);
});
```

Reply:
Well, yes, I think that is better, but eventually logger should also be passed as an argument instead of being global instance. Then we can create various scopes for different things to avoid passing extra info deep into function calls and maybe even have structured logging :roll_eyes: